README
# --
# README - readme file of OTRS
# Copyright (C) 2001-2012 OTRS AG, http://otrs.org/
# --
# $Id: README,v 1.47 2012/02/02 16:02:34 mg Exp $
# --
# This software comes with ABSOLUTELY NO WARRANTY. For details, see
# the enclosed file COPYING for license information (AGPL). If you
# did not receive this file, see http://www.gnu.org/licenses/agpl.txt.
# --

What is OTRS?
=============

  OTRS is an Open source Ticket Request System with many features to manage
  customer telephone calls and e-mails. It is distributed under the GNU
  AFFERO General Public License (AGPL) and tested on Linux, Solaris, AIX,
  Windows, FreeBSD, OpenBSD and Mac OS 10.x. Do you receive many e-mails and
  want to answer them with a team of agents? You're going to love OTRS!

  You can find a list of features at:
  http://doc.otrs.org/3.1/en/html/features-of-otrs.html


License:
========
  It is distributed under the AFFERO GNU General Public License - see the
  accompanying COPYING file for more details.

Documentation:
==============
  You can find quick documentation in README.* and the long version online at
  http://doc.otrs.org/.

Software requirements:
======================

  Perl
    * Perl 5.8.8 or higher

  Webserver
    * Webserver with CGI support (CGI is not recommended)
    * Apache2+mod_perl2 or higher (recommended, mod_perl is really fast!)
    * IIS6 or higher

  Databases
    * MySQL 4.1 or higher
    * PostgreSQL 8.0 or higher
    * Oracle 10g or higher
    * Microsoft SQL Server 2005 or higher

Directories & Files:
====================
$HOME (e. g. /opt/otrs/)
|
|  (all executables)
|--/bin/             (all system programs)
|   |--/otrs.PostMaster.pl      (email2db)
|   |--/otrs.PostMasterMail.pl  (email2db)
|   |--/cgi-bin/
|   |   |----- /index.pl        (Global Agent & Admin handle)
|   |   |----- /customer.pl     (Global Customer handle)
|   |   |----- /public.pl       (Global Public handle)
|   |   |----- /installer.pl    (Global Installer handle)
|   |   |----- /nph-genericinterface.pl (Global GenericInterface handle)
|   |--/fcgi-bin/               (If you're using FastCGI)
|
|  (all modules)
|--/Kernel/
|   |-----/Config.pm      (main configuration file)
|   |---- /Config/        (Configuration files)
|   |      |---- /Files/  (System generated, don't touch...)
|   |
|   |---- /Output/        (all output generating modules)
|   |      |---- /HTML/
|   |             |--- /Standard/*.dtl (all dtl files for Standard theme)
|   |
|   |--- /GenericInterface (GenericInterface framework)
|          |--- /Invoker/ (invoker backends)
|          |--- /Mapping/ (data mapping backends)
|          |--- /Operation/ (operation backends)
|          |--- /Transport/ (network transport backends)
|   |
|   |---- /Language/      (all translation files)
|   |
|   |---- /Modules/        (all action modules e. g. QueueView, Move, ...)
|   |      |----- /Admin*      (all modules for the admin interface)
|   |      |----- /Agent*      (all modules for the agent interface)
|   |      |----- /Customer*   (all modules for the customer interface)
|   |
|   |---- /System/         (back-end API modules, selection below)
|           |--- /Auth.pm        (authentication module)
|           |--- /AuthSession.pm (session authentication module)
|           |--- /DB.pm          (central DB interface)
|           |--- /DB/*.pm        (DB drivers)
|           |--- /DynamicField.pm (Interface to the DynamicField configuration)
|           |--- /DynamicField
|                 |--- /Backend.pm (Interface for using the dynamic fields)
|                 |--- /Backend/*.pm (DynamicField backend implementations)
|                 |--- /ObjectType/*.pm (DynamicField object type implementations)
|           |--- /Email.pm       (create and send e-mail)
|           |--- /EmailParser.pm (parsing e-mail)
|           |--- /GenericInterface/*.pm (all DB related GenericInterface modules)
|           |--- /Group.pm       (group module)
|           |--- /Log.pm         (log module)
|           |--- /Queue.pm       (information about queues. e. g. responses, ...)
|           |--- /Scheduler      (Scheduler files)
|                 |--- /TaskHandler/ (task handler backends for the Scheduler)
|           |--- /Ticket.pm      (ticket and article functions)
|           |--- /User.pm        (user module)
|           |--- /Request.pm    (HTTP/CGI abstraction module)
|
|  (data stuff)
|--/var/
|   |--/article/               (all incoming e-mails, plain 1/1 and all attachments ...
|   |                            ... separately (different files), if you want to store on disk)
|   |--/cron/                  (all cron jobs for escalations and such)
|   |
|   |--/fonts/                 (true type fonts for PDF generation)
|   |
|   |--/httpd/                 (all static files served by HTTP)
|   |   |--- /htdocs/
|   |         |--- /js/        (javascript files for OTRS)
|   |               |--- /js-cache/        (auto-generated minified JS files)
|   |               |--- /thirdparty/      (contains jQuery, CKEditor and other external JS libraries)
|   |         |--- /skins/     (CSS and images for front end)
|   |               |--- /Agent/        (Agent skins)
|   |                     |--- /default/ (default skin)
|   |                           |--- /css/ (stylesheets)
|   |                           |--- /css-cache/ (auto-generated minified CSS files)
|   |                           |--- /img/ (images)
|   |                     |--- /slim/    (additional skin)
|   |                           |--- /.../ (files)
|   |                     |--- /ivory/   (additional skin)
|   |                           |--- /.../ (files)
|   |               |--- /Customer/     (Customer skins)
|   |                     |--- /default/ (default skin)
|   |                           |--- /.../ (files)
|   |                     |--- /ivory/
|   |                           |--- /.../ (files)
|   |
|   |--/log/                   (log files)
|   |   |--/TicketCounter.log  (ticket counter)
|   |
|   |--/sessions/              (session info)
|   |
|   |--/spool/                 (spool files)
|   |
|   |--/stats/                 (statistics)
|   |
|   |--/tmp/                   (temporary files, such as cache)
|
|  (tools stuff)
|--/scripts/
|   |----  /database/
|           |--- /otrs-schema.(mysql|postgresql|*).sql (create database script)
|           |--- /otrs-initial_insert.(mysql|postgresql|*).sql (all initial sql data - e. g.
|           |                                                   root user, queues, ...)
|           |--- /otrs-schema-post.(mysql|postgresql|*).sql (create foreign keys script)
|


CHANGES
# --
# CHANGES - change log of OTRS
# Copyright (C) 2001-2012 OTRS AG, http://otrs.org/
# --
# $Id: CHANGES,v 1.2201 2012/03/09 13:00:30 mg Exp $
# --
# This software comes with ABSOLUTELY NO WARRANTY. For details, see
# the enclosed file COPYING for license information (AGPL). If you
# did not receive this file, see http://www.gnu.org/licenses/agpl.txt.
# --

3.1.3 2012-??-??
 - 2012-03-09 Fixed bug#8286 - Adressbook forgets selected entries.
 - 2012-03-07 Fixed bug#8297 - Selectbox for new owner causes ajax update
    on every change made via keyboard.
 - 2012-03-07 Updated Catalan translation, thanks to Antionio Linde!
 - 2012-03-05 Fixed bug#7545 - AgentTicketBounce lacks permission checks.

3.1.2 2012-03-06
 - 2012-03-01 Fixed bug#8282 - Dropdown and Multiselect Dynamic Fields without Possible values
    causes errors with LayoutObject BuildSelection function.
 - 2012-03-01 Fixed bug#8277 - DynamicField Values not deleted when ticket is deleted.
 - 2012-02-28 Fixed bug#8274 - Dynamic Fields ACLs does not work correctly at Ticket Split.
 - 2012-02-28 Improved #7526 - Fixed handling of special characters (, ), &, - within statistics.
 - 2012-02-27 Fixed bug#8255 - TicketSearch in DynamicFields doesn't support ExtendedSearchCondition.
 - 2012-02-25 Fixed bug#8258 - DynamicField date value is reset to NULL.
 - 2012-02-23 Fixed bug#8214 - OTRS Init script on Red Hat / SUSE fails to check scheduler.
 - 2012-02-23 Fixed bug#8227 - LDAP user syncronisation doesn't work.
 - 2012-02-23 Fixed bug#8235 - Searching on DynamicFields, results are lost on next page.
 - 2012-02-21 Fixed bug#8252 - Small typo in German translation of AgentTicketZoom.
 - 2012-02-16 Fixed bug#8226 - Problem with Customized DynamicFields in TicketOverviewSmall.
 - 2012-02-16 Fixed bug#8233 - DB error migrating stats on Oracle.
 - 2012-02-14 Fixed bug#8224 - Admin Responses screen does not allow to filter responses.
 - 2012-02-14 Fixed bug#7652 - OpenSearch providers are served with wrong mime type.
    Follow-up fix.
 - 2012-02-14 Fixed bug#8140 - Shortcut for creating new e-mail ticket doesn't work.
    The hotkey for "new email ticket" was changed from e to m to avoid a collision.
 - 2012-02-14 Fixed bug#8144 - Typo and improved logging in GenericAgent.pm.
 - 2012-02-14 Fixed bug#8183 - Canceling AgentTicketCompose on an unlocked ticket generates
    ChallengeToken error.
 - 2012-02-13 Fixed bug#8219 - TicketCreate and TicketUpdate GI ticket operations requires a valid
    CustomerUser.
 - 2012-02-13 Fixed bug#8201 - Popup in browser always open on leftmost display when using
    dual monitor setup.
 - 2012-02-13 Fixed bug#8202 - Kernel::System::JSON->Decode() dies when providing malformed JSON.
 - 2012-02-13 Fixed bug#8211 - Typos in Labels: DYNAMICFILED.
 - 2012-02-09 Fixed bug#7109 - Statistics error when PDF support is not active.

3.1.1 2012-02-14
 - 2012-02-09 Fixed bug#8184 - Uninstall and upgrade fails if file in package can't be removed.
 - 2012-02-09 Fixed bug#8199 - Linked tickets open only in tabs.
 - 2012-02-03 Fixed bug#8148 - Wrong presentation of queue structure in CustomerTicketMessage.
 - 2012-02-03 Fixed bug#8137 - Issues with Owner list refresh when selecting a new Owner.
 - 2012-02-03 Fixed bug#8180 - bin/otrs.LoaderCache.pl exit code is wrong.

3.1.0.rc1 2012-02-07
 - 2012-02-03 Fixed bug#8171 - Table rows have different heights.
 - 2012-02-02 Fixed bug#7937 - HTMLUtils.pm ignore to much of e-mail source code.
 - 2012-02-02 Fixed bug#7972 - Some mails may not present HTML part when using rich viewing.
 - 2012-02-02 Fixed bug#8179 - DynamicField backend DateTime renders timestamps with seconds.
 - 2012-01-31 Fixed bug#8163 - Ticket / Article title can't be copied if value is too long.
 - 2012-01-31 Fixed bug#8161 - History hover broken - missing title attribute.
 - 2012-01-30 Fixed bug#8147 - Possible values for queues not being considered in ACLs.
 - 2012-01-30 Fixed bug#8021 - Creating new Dynamic Field is not adding the field name
    to the X-OTRS header lists configuration for the PostMaster filters.
 - 2012-01-27 Fixed bug#8145 - TicketFreeTime entries in search profiles don't get updated.
 - 2012-01-27 Fixed bug#2820 - Wide character in Syslog message causes Perl crash on utf8 systems.

3.1.0.beta5 2012-01-31
 - 2012-01-25 Fixed bug#7890 - Changed wording of config setting for RFC 5321 compliance.
 - 2012-01-24 Fixed bug#8068 - Corrected field & DynamicField preselection on TicketSplit.
 - 2012-01-24 Fixed bug#8115 - Richtext editor not show in customer interface after switching
    to Catalan frontend language.
 - 2012-01-23 Fixed bug#7994 - ACL: action-restrictions not possible for all ticket actions.
 - 2012-01-23 Fixed bug#7984 - Unable to select the output format of statistics.
 - 2012-01-23 Fixed bug#8136 - Browser Time Detection and time zone UTC generates warning in
    web server error log.
 - 2012-01-23 Fixed bug#8132 - Browser version message misleading when 'compatibility mode'
    is enabled in Internet Explorer.
 - 2012-01-23 Fixed bug#7916 - Address Book doesn't work correctly.
 - 2012-01-23 Fixed bug#8019 - Ticket customer info widget has unneeded scroll bars.
 - 2012-01-23 Fixed bug#8066 - TicketAcl with empy possible List for DynamicFields crashes.
 - 2012-01-20 Fixed bug#7495 - Cursor stands still in editor in IE 9.
 - 2012-01-20 Fixed bug#8129 - Ticket Unwatch may lead to errorscreen.
 - 2012-01-20 Updated Finnish translation file, thanks to Mikko Hynninen!
 - 2012-01-17 Fixed bug#8117 - Can't create ticket for newly created customer user.
 - 2012-01-16 Fixed bug#8094 - Typo In Ticket.pm.
 - 2012-01-16 Creation of QueueObject was not possible because of missing EncodeObject in CustomerUserGenericTicket.pm.

3.1.0.beta4 2012-01-17
 - 2012-01-12 Fixed bug#8107 - Ticket state is not set as default on ticket edit screens.
 - 2012-01-12 Fixed bug#8105 - Changing Priority does not update all relevant SysConfig fields.

Kernel/Config.pm
# --
# Kernel/Config.pm - Config file for OTRS kernel
# Copyright (C) 2001-2010 OTRS AG, http://otrs.org/
# --
# $Id: Config.pm.dist,v 1.23 2010/01/13 22:25:00 martin Exp $
# --
# This software comes with ABSOLUTELY NO WARRANTY. For details, see
# the enclosed file COPYING for license information (AGPL). If you
# did not receive this file, see http://www.gnu.org/licenses/agpl.txt.
# --
#  Note:
#
#  -->> OTRS does have a lot of config settings. For more settings
#       (Notifications, Ticket::ViewAccelerator, Ticket::NumberGenerator,
#       LDAP, PostMaster, Session, Preferences, ...) see
#       Kernel/Config/Defaults.pm and copy your wanted lines into "this"
#       config file. This file will not be changed on update!
#
# --

package Kernel::Config;
use strict;
use warnings;

my $myotrsroot= '/home/mdupont/projects/otrs/git/public/otrs';

our $dbuser='otrs';
our $dbpass='5ypUVX4jqK1g';
our $basepath='';
our $dbname='otrs2';
our $dbserver='';
our $dbport='5432';
our $dbtype='pgsql';
our $dsn = "DBI:Pg:dbname";

sub Load {
    my $Self = shift;
    # ---------------------------------------------------- #
    # ---------------------------------------------------- #
    #                                                      #
    #         Start of your own config options!!!          #
    #                                                      #
    # ---------------------------------------------------- #
    # ---------------------------------------------------- #

    # ---------------------------------------------------- #
    # database settings                                    #
    # ---------------------------------------------------- #
    # DatabaseHost
    # (The database host.)
    $Self->{DatabaseHost} = $dbserver || 'localhost';
    # Database
    # (The database name.)
    $Self->{Database} = $dbname || 'otrs';
    # DatabaseUser
    # (The database user.)
    $Self->{DatabaseUser} = $dbuser || 'otrs';
    # DatabasePw
    # (The password of database user. You also can use bin/otrs.CryptPassword.pl
    # for crypted passwords.)
    $Self->{DatabasePw} = $dbpass;
    # DatabaseDSN
    # (The database DSN for MySQL ==> more: "man DBD::mysql")
    $Self->{DatabaseDSN} =  "$dsn=$Self->{Database};host=$Self->{DatabaseHost};port=$dbport;";

    # (The database DSN for PostgreSQL ==> more: "man DBD::Pg")
    # if you want to use a local socket connection
#    $Self->{DatabaseDSN} = "DBI:Pg:dbname=$Self->{Database};";
    # if you want to use a tcpip connection
#    $Self->{DatabaseDSN} = "DBI:Pg:dbname=$Self->{Database};host=$Self->{DatabaseHost};";

    # ---------------------------------------------------- #
    # fs root directory
    # ---------------------------------------------------- #
    $Self->{Home} = $myotrsroot; #'/usr/share/otrs';

    # ---------------------------------------------------- #
    # insert your own config settings "here"               #
    # config settings taken from Kernel/Config/Defaults.pm #
    # ---------------------------------------------------- #
    # $Self->{SessionUseCookie} = 0;
    # $Self->{CheckMXRecord} = 0;

    # ---------------------------------------------------- #
    # switch off the web based installer for the Debian package
    $Self->{SecureMode} = 1;

    # ---------------------------------------------------- #
    # data inserted by installer                           #
    # ---------------------------------------------------- #
    # $DIBI$
    $Self->{PostmasterUserID} = "1";
    

$Self->{'PostmasterX-Header'} =  [
  'From',
  'To',
  'Cc',
  'Reply-To',
  'ReplyTo',
  'Subject',
  'Message-ID',
  'Message-Id',
  'Resent-To',
  'Resent-From',
  'Precedence',
  'Mailing-List',
  'List-Id',
  'List-Archive',
  'Errors-To',
  'References',
  'In-Reply-To',
  'X-Loop',
  'X-Spam-Flag',
  'X-Spam-Status',
  'X-Spam-Level',
  'X-No-Loop',
  'X-Priority',
  'Importance',
  'X-Mailer',
  'User-Agent',
  'Organization',
  'X-Original-To',
  'Delivered-To',
  'Envelope-To',
  'Return-Path',
  'X-OTRS-Loop',
  'X-OTRS-Info',
  'X-OTRS-Priority',
  'X-OTRS-Queue',
  'X-OTRS-Lock',
  'X-OTRS-Ignore',
  'X-OTRS-State',
  'X-OTRS-State-PendingTime',
  'X-OTRS-Type',
  'X-OTRS-Service',
  'X-OTRS-SLA',
  'X-OTRS-CustomerNo',
  'X-OTRS-CustomerUser',
  'X-OTRS-SenderType',
  'X-OTRS-ArticleType',
  'X-OTRS-FollowUp-Priority',
  'X-OTRS-FollowUp-Queue',
  'X-OTRS-FollowUp-Lock',
  'X-OTRS-FollowUp-State',
  'X-OTRS-FollowUp-State-PendingTime',
  'X-OTRS-FollowUp-Type',
  'X-OTRS-FollowUp-Service',
  'X-OTRS-FollowUp-SLA',
  'X-OTRS-FollowUp-SenderType',
  'X-OTRS-FollowUp-ArticleType'
];

$Self->{'CustomerPreferences'} =  {
  'Module' => 'Kernel::System::CustomerUser::Preferences::DB',
  'Params' => {
    'Table' => 'customer_preferences',
    'TableKey' => 'preferences_key',
    'TableUserID' => 'user_id',
    'TableValue' => 'preferences_value'
  }
};

#    $Self->{"Ticket::ViewableStateType"} = ["TicketViewableStateType"];
$Self->{'Ticket::ViewableStateType'} =  [
  'new',
  'open',
  'pending reminder',
  'pending auto'
];

#    $Self->{"Ticket::UnlockStateType"} = "TicketUnlockStateType";
$Self->{'Ticket::UnlockStateType'} =  [
  'new',
  'open'
];

$Self->{'Ticket::ViewableLocks'} =  [
  '\'unlock\'',
  '\'tmp_lock\''
];

    $Self->{"ArticleDir"}=$myotrsroot . "/var/article";

#    $Self->{"Ticket::EventModulePost"} = {TestModule => {
#	test =>1,
#	Module => "TestPostEventModule"					  }};

$Self->{'Ticket::EventModulePost'}->{'98-ArticleSearchIndex'} =  {
  'Event' => '(ArticleCreate|ArticleUpdate)',
  'Module' => 'Kernel::System::Ticket::Event::ArticleSearchIndex'
};


$Self->{'PostMaster::PreFilterModule'}->{'1-SystemMonitoring'} =  {
  'ArticleType' => 'note-report',
  'CloseActionState' => 'closed successful',
  'ClosePendingTime' => '172800',
  'CloseTicketRegExp' => 'OK|UP',
  'DefaultService' => 'Host',
  'FreeTextHost' => '12',
  'FreeTextService' => '9',
  'FreeTextState' => '2',
  'FromAddressRegExp' => 'nagios@example.com',
  'HostRegExp' => '\\s*Host:\\s+(.*)\\s*',
  'Module' => 'Kernel::System::PostMaster::Filter::SystemMonitoring',
  'NewTicketRegExp' => 'CRITICAL|DOWN',
  'SenderType' => 'system',
  'ServiceRegExp' => '\\s*Service:\\s+(.*)\\s*',
  'StateRegExp' => '\\s*State:\\s+(\\S+)'
};

#    use Kernel::System::DynamicField::Backend;
#    $Self->{'DynamicFields::Backend'} = {
#	"FieldType" => {
#	    "Module" => "TestFieldTypeModule"
#	}
#    };


$Self->{'iPhone::Frontend::AgentTicketMove'}->{'ArticleFreeText'} =  {
  '1' => '0',
  '2' => '0',
  '3' => '0'
};
$Self->{'iPhone::Frontend::AgentTicketMove'}->{'TicketFreeTime'} =  {
  '1' => '0',
  '2' => '0',
  '3' => '0',
  '4' => '0',
  '5' => '0',
  '6' => '0'
};
$Self->{'iPhone::Frontend::AgentTicketMove'}->{'TicketFreeText'} =  {
  '1' => '0',
  '10' => '0',
  '11' => '0',
  '12' => '0',
  '13' => '0',
  '14' => '0',
  '15' => '0',
  '16' => '0',
  '2' => '0',
  '3' => '0',
  '4' => '0',
  '5' => '0',
  '6' => '0',
  '7' => '0',
  '8' => '0',
  '9' => '0'
};
$Self->{'iPhone::Frontend::AgentTicketMove'}->{'TimeUnits'} =  '1';
$Self->{'iPhone::Frontend::AgentTicketMove'}->{'StateDefault'} =  'open';
$Self->{'iPhone::Frontend::AgentTicketMove'}->{'StateType'} =  [
  'open',
  'closed'
];
$Self->{'iPhone::Frontend::AgentTicketMove'}->{'State'} =  '1';
$Self->{'iPhone::Frontend::AgentTicketMove'}->{'Owner'} =  '1';
$Self->{'iPhone::Frontend::AgentTicketMove'}->{'RequiredLock'} =  '1';
$Self->{'iPhone::Frontend::AgentTicketMove'}->{'Permission'} =  'compose';
$Self->{'iPhone::Frontend::AgentTicketCompose'}->{'ArticleFreeText'} =  {
  '1' => '0',
  '2' => '0',
  '3' => '0'
};
$Self->{'iPhone::Frontend::AgentTicketCompose'}->{'TicketFreeTime'} =  {
  '1' => '0',
  '2' => '0',
  '3' => '0',
  '4' => '0',
  '5' => '0',
  '6' => '0'
};
$Self->{'iPhone::Frontend::AgentTicketCompose'}->{'TicketFreeText'} =  {
  '1' => '0',
  '10' => '0',
  '11' => '0',
  '12' => '0',
  '13' => '0',
  '14' => '0',
  '15' => '0',
  '16' => '0',
  '2' => '0',
  '3' => '0',
  '4' => '0',
  '5' => '0',
  '6' => '0',
  '7' => '0',
  '8' => '0',
  '9' => '0'
};
$Self->{'iPhone::Frontend::AgentTicketCompose'}->{'TimeUnits'} =  '1';
$Self->{'iPhone::Frontend::ComposeExcludeCcRecipients'} =  '0';
$Self->{'iPhone::Frontend::ComposeReplaceSenderAddress'} =  '0';
$Self->{'iPhone::Frontend::ComposeAddCustomerAddress'} =  '1';
$Self->{'iPhone::Frontend::AgentTicketCompose'}->{'StateType'} =  [
  'open',
  'closed',
  'pending auto',
  'pending reminder'
];
$Self->{'iPhone::Frontend::AgentTicketCompose'}->{'StateDefault'} =  'open';
$Self->{'iPhone::Frontend::AgentTicketCompose'}->{'State'} =  '1';
$Self->{'iPhone::Frontend::AgentTicketCompose'}->{'RequiredLock'} =  '1';
$Self->{'iPhone::Frontend::AgentTicketCompose'}->{'Permission'} =  'compose';
$Self->{'iPhone::Frontend::AgentTicketClose'}->{'HistoryComment'} =  '%%Close';
$Self->{'iPhone::Frontend::AgentTicketClose'}->{'HistoryType'} =  'AddNote';
$Self->{'iPhone::Frontend::AgentTicketClose'}->{'ArticleFreeText'} =  {
  '1' => '0',
  '2' => '0',
  '3' => '0'
};
$Self->{'iPhone::Frontend::AgentTicketClose'}->{'TicketFreeTime'} =  {
  '1' => '0',
  '2' => '0',
  '3' => '0',
  '4' => '0',
  '5' => '0',
  '6' => '0'
};
$Self->{'iPhone::Frontend::AgentTicketClose'}->{'TicketFreeText'} =  {
  '1' => '0',
  '10' => '0',
  '11' => '0',
  '12' => '0',
  '13' => '0',
  '14' => '0',
  '15' => '0',
  '16' => '0',
  '2' => '0',
  '3' => '0',
  '4' => '0',
  '5' => '0',
  '6' => '0',
  '7' => '0',
  '8' => '0',
  '9' => '0'
};
$Self->{'iPhone::Frontend::AgentTicketClose'}->{'TimeUnits'} =  '1';
$Self->{'iPhone::Frontend::AgentTicketClose'}->{'Title'} =  '0';
$Self->{'iPhone::Frontend::AgentTicketClose'}->{'ArticleTypes'} =  {
  'note-external' => '0',
  'note-internal' => '1',
  'note-report' => '0'
};
$Self->{'iPhone::Frontend::AgentTicketClose'}->{'ArticleTypeDefault'} =  'note-internal';
$Self->{'iPhone::Frontend::AgentTicketClose'}->{'Body'} =  '';
$Self->{'iPhone::Frontend::AgentTicketClose'}->{'Subject'} =  'Close';
$Self->{'iPhone::Frontend::AgentTicketClose'}->{'Note'} =  '1';
$Self->{'iPhone::Frontend::AgentTicketClose'}->{'StateDefault'} =  'closed successful';
$Self->{'iPhone::Frontend::AgentTicketClose'}->{'StateType'} =  [
  'closed'
];
$Self->{'iPhone::Frontend::AgentTicketClose'}->{'State'} =  '1';
$Self->{'iPhone::Frontend::AgentTicketClose'}->{'Responsible'} =  '0';
$Self->{'iPhone::Frontend::AgentTicketClose'}->{'Owner'} =  '0';
$Self->{'iPhone::Frontend::AgentTicketClose'}->{'Service'} =  '0';
$Self->{'iPhone::Frontend::AgentTicketClose'}->{'TicketType'} =  '0';
$Self->{'iPhone::Frontend::AgentTicketClose'}->{'RequiredLock'} =  '1';
$Self->{'iPhone::Frontend::AgentTicketClose'}->{'Permission'} =  'close';
$Self->{'iPhone::Frontend::AgentTicketNote'}->{'HistoryComment'} =  '%%Note';
$Self->{'iPhone::Frontend::AgentTicketNote'}->{'HistoryType'} =  'AddNote';
$Self->{'iPhone::Frontend::AgentTicketNote'}->{'ArticleFreeText'} =  {
  '1' => '0',
  '2' => '0',
  '3' => '0'
};
$Self->{'iPhone::Frontend::AgentTicketNote'}->{'TicketFreeTime'} =  {
  '1' => '0',
  '2' => '0',
  '3' => '0',
  '4' => '0',
  '5' => '0',
  '6' => '0'
};
$Self->{'iPhone::Frontend::AgentTicketNote'}->{'TicketFreeText'} =  {
  '1' => '0',
  '10' => '0',
  '11' => '0',
  '12' => '0',
  '13' => '0',
  '14' => '0',
  '15' => '0',
  '16' => '0',
  '2' => '0',
  '3' => '0',
  '4' => '0',
  '5' => '0',
  '6' => '0',
  '7' => '0',
  '8' => '0',
  '9' => '0'
};
$Self->{'iPhone::Frontend::AgentTicketNote'}->{'TimeUnits'} =  '1';
$Self->{'iPhone::Frontend::AgentTicketNote'}->{'Title'} =  '0';
$Self->{'iPhone::Frontend::AgentTicketNote'}->{'ArticleTypes'} =  {
  'note-external' => '1',
  'note-internal' => '1',
  'note-report' => '0'
};
$Self->{'iPhone::Frontend::AgentTicketNote'}->{'ArticleTypeDefault'} =  'note-internal';
$Self->{'iPhone::Frontend::AgentTicketNote'}->{'Body'} =  '';
$Self->{'iPhone::Frontend::AgentTicketNote'}->{'Subject'} =  'Note';
$Self->{'iPhone::Frontend::AgentTicketNote'}->{'Note'} =  '1';
$Self->{'iPhone::Frontend::AgentTicketNote'}->{'StateType'} =  [
  'open',
  'closed',
  'pending reminder',
  'pending auto'
];
$Self->{'iPhone::Frontend::AgentTicketNote'}->{'State'} =  '0';
$Self->{'iPhone::Frontend::AgentTicketNote'}->{'Responsible'} =  '0';
$Self->{'iPhone::Frontend::AgentTicketNote'}->{'Owner'} =  '0';
$Self->{'iPhone::Frontend::AgentTicketNote'}->{'Service'} =  '0';
$Self->{'iPhone::Frontend::AgentTicketNote'}->{'TicketType'} =  '0';
$Self->{'iPhone::Frontend::AgentTicketNote'}->{'RequiredLock'} =  '0';
$Self->{'iPhone::Frontend::AgentTicketNote'}->{'Permission'} =  'note';



$Self->{'iPhone::Frontend::AgentTicketPhone'}->{'HistoryComment'} =  '';
$Self->{'iPhone::Frontend::AgentTicketPhone'}->{'HistoryType'} =  'PhoneCallCustomer';
$Self->{'iPhone::Frontend::AgentTicketPhone'}->{'ArticleFreeText'} =  {
  '1' => '0',
  '2' => '0',
  '3' => '0'
};
$Self->{'iPhone::Frontend::AgentTicketPhone'}->{'TicketFreeTime'} =  {
  '1' => '0',
  '2' => '0',
  '3' => '0',
  '4' => '0',
  '5' => '0',
  '6' => '0'
};
$Self->{'iPhone::Frontend::AgentTicketPhone'}->{'TicketFreeText'} =  {
  '1' => '0',
  '10' => '0',
  '11' => '0',
  '12' => '0',
  '13' => '0',
  '14' => '0',
  '15' => '0',
  '16' => '0',
  '2' => '0',
  '3' => '0',
  '4' => '0',
  '5' => '0',
  '6' => '0',
  '7' => '0',
  '8' => '0',
  '9' => '0'
};
$Self->{'iPhone::Frontend::AgentTicketPhone'}->{'TimeUnits'} =  '1';
$Self->{'iPhone::Frontend::AgentTicketPhone'}->{'SenderType'} =  'customer';
$Self->{'iPhone::Frontend::AgentTicketPhone'}->{'ArticleTypeDefault'} =  'phone';
$Self->{'iPhone::Frontend::AgentTicketPhone'}->{'PriorityDefault'} =  '3 normal';
$Self->{'iPhone::Frontend::AgentTicketPhone'}->{'StateDefault'} =  'open';
$Self->{'iPhone::Frontend::AgentTicketPhone'}->{'State'} =  '1';
$Self->{'iPhone::Frontend::AgentTicketPhone'}->{'CustomerID'} =  '1';
$Self->{'iPhone::Frontend::AgentTicketPhone'}->{'Responsible'} =  '1';
$Self->{'iPhone::Frontend::AgentTicketPhone'}->{'Owner'} =  '1';
$Self->{'iPhone::Frontend::AgentTicketPhone'}->{'Service'} =  '1';

$Self->{'iPhone::Frontend::AgentTicketPhone'}->{'TicketType'} =  '1';

$Self->{'iPhone::Frontend::AgentTicketPhone'}->{'Permission'} =  'compose';

$Self->{'iPhone::LogFile'} =  '/tmp/iPhone.log';
$Self->{'iPhone::DebugLog'} =  '0';


    # ---------------------------------------------------- #
    # ---------------------------------------------------- #
    #                                                      #
    #           End of your own config options!!!          #
    #                                                      #
    # ---------------------------------------------------- #
    # ---------------------------------------------------- #
}


# ---------------------------------------------------- #
# needed system stuff (don't edit this)                #
# ---------------------------------------------------- #
use strict;
use warnings;

use vars qw(@ISA $VERSION);
$VERSION = qw($Revision: 1.23 $)[1];

use Kernel::Config::Defaults;
push (@ISA, 'Kernel::Config::Defaults');

# -----------------------------------------------------#

1;
